Segment-ComScore/Classes/SEGComScoreIntegration.m

```diff
@@ -500,47 +476,85 @@ - (void)videoAdStarted:(NSDictionary *)properties withOptions:(NSDictionary *)in
     // StreamingAnalytics's asset. This is because ns_st_ci will have already been set via asset_id in a
     // Content Started calls (if this is a mid or post-roll), or via content_asset_id on Video Playback
     // Started (if this is a pre-roll).
-    NSString *contentId = [[[self.streamAnalytics playbackSession] asset] label:@"ns_st_ci"] ?: @"0";
+    NSDictionary *labels = self.streamAnalytics.configuration.labels;
+    NSString *contentId = [labels objectForKey:@"ns_st_ci"] ?: @"0";
+
     NSMutableDictionary *mapWithContentId = [NSMutableDictionary dictionaryWithDictionary:map];
     [mapWithContentId setValue:contentId forKey:@"ns_st_ci"];
 
-    [[self.streamAnalytics playbackSession] setAssetWithLabels:mapWithContentId];
-    SEGLog(@"[[SCORStreamingAnalytics playbackSession] setAssetWithLabels:%@", mapWithContentId);
+    SCORStreamingContentMetadata *contentMetadata = [self instantiateContentMetaData:mapWithContentId];
+    NSString *adType = [properties valueForKey:@"type"];
+    __block NSInteger setMediaType;
+    if ([adType isEqualToString:@"pre-roll"]) {
+        setMediaType = SCORStreamingAdvertisementTypeBrandedOnDemandPreRoll;
+    } else if ([adType isEqualToString:@"midroll"]) {
+        setMediaType = SCORStreamingAdvertisementTypeBrandedOnDemandMidRoll;
+    } else if ([adType isEqualToString:@"post-roll"]) {
+        setMediaType = SCORStreamingAdvertisementTypeBrandedOnDemandPostRoll;
+    } else {
+        setMediaType = SCORStreamingAdvertisementTypeBrandedOnDemandPreRoll;
```

Comment:
Why do we default to pre-roll here?

Reply:
Updated that to be `SCORStreamingAdvertisementTypeOther`